stroyprombeton/tests/tests_views.py

```diff
@@ -797,22 +797,22 @@ def test_too_many_tags_collapse(self):
         self.assertIn(f'от {from_index} м до {to_index - 1} м', tags_text)
         self.assertGreater(product.options.first().tags.count(), settings.TAGS_UI_LIMIT)
 
-    # @todo #495:30m  Resurrect `test_filter_options_by_tags`.
-    #  Seems response parsing is needed here.
-    @unittest.expectedFailure
     def test_filter_options_by_tags(self):
         """Category page should not contain options, excluded by tags selection."""
-        tag_slug = '2-m'
-        tag_qs = models.Tag.objects.filter(slug=tag_slug)
+        tag_qs = models.Tag.objects.filter_by_options(
+            options=models.Option.objects.filter_descendants(self.root_category)
+        )
         tag = tag_qs.first()
-        response = self.get_category_page(category=self.root_category, tags=tag_qs)
+        tag_qs = models.Tag.objects.filter(slug=tag.slug)
+        soup = self.get_category_soup(category=self.root_category, tags=tag_qs[1:])
+        options = soup.find(class_='product-list').find_all(class_='table-link')
+        returned = {o.text for o in options}
 
-        disappeared = (
+        disappeared_qs = (
             models.Option.objects
             .bind_fields().active()
             .filter_descendants(self.root_category)
             .exclude(tags=tag)
         )
-        self.assertFalse(
-            any(str(o.id) in response.content.decode() for o in disappeared)
-        )
+        disappeared = {f'{d.name} {d.mark}' for d in disappeared_qs}
```

Comment:
Let's create a todo to implement `catalog_name` property for Product, that will print `{d.name} {d.mark}`